### ISSUE TITLE
[FR] Support for multiple/headless simulators through Bluepill or others

### ISSUE THREAD

**Issue by samskiter** [fastlane/fastlane#9216]
### Issue Description

This new library has just popped up, supposedly allowing you to run multiple simulators and even run them headless: https://github.com/linkedin/bluepill

Doing this would allow for much more efficient testing on one machine. Rather than using build-for-test to build on one machine, then distributing those artefacts to other machines for actual running (because of the one-simulator-per-machine problem), with this you could build and run all your targets from one machine in parallel.

**Comment by samskiter** [comment]
A little more information about how the tools works:

First you build using build-for-testing. Then you manually find the generated .app and TestScheme files and pass those to bluepill.

Bluepill can read the schemes and tell you the tests to run, it can automatically retry failing tests and a bunch of other cool stuff

Additional nice-to-haves:

Auto sharding of the tests - by listing the tests & dividing them in the groups we could provide automatic behaviour to divide tests between a number of simulators (specified as a variable) and so save time on slow tests like UI tests.

**Comment by PetrosIchor** [comment]
https://github.com/tbrand/fastlane-plugin-bluepill

https://github.com/Shashikant86/fastlane-plugin-bluepillar

**Comment by PetrosIchor** [comment]
https://github.com/tbrand/fastlane-plugin-bluepill

https://github.com/Shashikant86/fastlane-plugin-bluepillar

**Comment by KrauseFx** [comment]
That's awesome, thanks for sharing @PetrosIchor, I didn't know about these plugins 👍

**Comment by davetobin** [comment]
Xcode 9 includes running of tests on multiple simulators in parallel. Will this be supported in Fastlane?

**Comment by mpirri** [comment]
@davetobin That is definitely in our plans! We are working on a change like this for `shapshot` and screenshots (#9570), and a similar change for `scan` would be awesome as well!  🚀
